keras/losses/losses.py

```diff
@@ -1934,3 +1934,76 @@ def ctc(y_true, y_pred):
     return ops.ctc_loss(
         y_true, y_pred, label_length, input_length, mask_index=0
     )
+
+
+@keras_export("keras.losses.Dice")
+class Dice(LossFunctionWrapper):
+    """Computes the Dice loss value between `y_true` and `y_pred`.
+
+    Formula:
+    ```python
+    loss = 1 - (2 * sum(y_true * y_pred)) / (sum(y_true) + sum(y_pred))
+    ```
+
+    Args:
+        y_true: tensor of true targets.
+        y_pred: tensor of predicted targets.
+        smooth: coefficient to avoid division by zero when both `y_true`
+            and `y_pred` do not contain any foreground pixels.
+
+    Returns:
+        Dice loss value.
+    """
+
+    def __init__(
+        self,
+        smooth=1e-6,
+        reduction="sum_over_batch_size",
+        name="dice",
+    ):
+        super().__init__(
+            dice,
+            name=name,
+            reduction=reduction,
+            smooth=smooth,
+        )
+        self.smooth = smooth
+
+    def get_config(self):
+        return {
+            "name": self.name,
+            "reduction": self.reduction,
+            "smooth": self.smooth,
+        }
+
+
+@keras_export("keras.losses.dice")
+def dice(y_true, y_pred, smooth=1e-6):
+    """Computes the Dice loss value between `y_true` and `y_pred`.
+
+    Formula:
+    ```python
+    loss = 1 - (2 * sum(y_true * y_pred)) / (sum(y_true) + sum(y_pred))
+    ```
+
+    Args:
+        y_true: tensor of true targets.
+        y_pred: tensor of predicted targets.
+        smooth: coefficient to avoid division by zero when both `y_true`
+            and `y_pred` do not contain any foreground pixels.
+
+    Returns:
+        Dice loss value.
+    """
+    y_true = ops.cast(y_true, dtype="float32")
```

Comment:
There's no need to force the use of `float32`, you can just use `ops.convert_to_tensor(y_true)`